src/components/advanced-form/advanced-form.jsx

```diff
@@ -0,0 +1,179 @@
+  id,
+  labelText,
+  password = false,
+  placeholder,
+  isValid,
+}) => {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext();
+  return (
+    <>
+      <div className="flex font-bold font-arial flex-col py-3">
+        <label htmlFor={id} className={`mb-1.5 text-lg text-start`}>
+          {labelText}
+        </label>
+
+        <input
+          {...register(id)}
+          className={`font-normal bg-background py-3 px-4 border-solid rounded-none border ${
+            isValid ? "border-black" : "border-danger focus:outline-danger"
+          } `}
+          type={password ? "password" : "text"}
+          placeholder={placeholder || "Type here"}
+          errors={errors}
+        />
+      </div>
+    </>
+  );
+};
+
+AdvancedForm.DropzoneField = ({ name, multiple, className, labelText, id }) => {
+  const { control } = useFormContext();
+
+  return (
+    <Controller
+      render={({ field: { onChange } }) => (
+        <Dropzone
+          multiple={multiple}
+          onChange={(e) =>
+            onChange(multiple ? e.target.files : e.target.files?.[0] ?? null)
+          }
+          className={className}
+          labelText={labelText}
+          id={id}
+        />
+      )}
+      name={name}
+      control={control}
+      defaultValue=""
+    />
+  );
+};
+
+const Dropzone = ({ multiple, onChange, className, labelText, id }) => {
+  const [files, setFiles] = useState([]);
+  const onDrop = (acceptedFiles) => {
+    setFiles((files) => files.concat(...acceptedFiles));
+  };
+  const { getRootProps, getInputProps } = useDropzone({ onDrop });
+
+  return (
+    <>
+      <label
+        htmlFor={id}
+        className={`mb-1.5 text-lg text-start font-bold font-arial`}
+      >
+        {labelText}
+      </label>
+      <div {...getRootProps()} className={`border-solid border-2 ${className}`}>
+        <input {...getInputProps({ onChange })} id={id} />
+        <p>Drop the files here ...</p>
+      </div>
+      <div>
+        Files :
+        {files.map((file) => (
+          <div key={file.path}>{file.path} </div>
+        ))}
+      </div>
+    </>
+  );
+};
+
+AdvancedForm.Slider = ({ className, id, labelText }) => {
+  const { register } = useFormContext();
+  return (
+    <>
+      <div className="flex font-bold font-arial flex-col py-3">
+        <label htmlFor={id} className={`mb-1.5 text-lg text-start`}>
+          {labelText}
+        </label>
+
+        <input {...register(id)} className={`${className}`} type="range" />
+      </div>
+    </>
+  );
+};
+
+AdvancedForm.checkbox = ({ className, id, labelText }) => {
+  const { register } = useFormContext();
+  return (
+    <>
+      <div className="font-normal font-arial py-3">
+        <input {...register(id)} className={`${className}`} type="checkbox" />
+        <label htmlFor={id} className={`ml-1.5 text-lg text-start `}>
+          {labelText}
+        </label>
+      </div>
+    </>
+  );
+};
+
+AdvancedForm.Dropdown = ({
```

Comment:
I am unsure of how to set up some of the components, such as the dropdown, to show data and check functionality. Maybe include comments with how to integrate them.